### DISCUSSION TITLE
React Native component types as options

### DISCUSSION THREAD

**Discussion by zakharchenkoAndrii** [storybookjs/react-native#376] [Ideas]
It would be super useful to have a common exported package converting RN core library types to the array. To be used as options.
As an example:
RN `TextInput` `ReturnKeyType` is defined as type:
```
export type ReturnKeyType = 'done' | 'go' | 'next' | 'search' | 'send';
export type ReturnKeyTypeAndroid = 'none' | 'previous';
export type ReturnKeyTypeIOS = 'default' | 'google' | 'join' | 'route' | 'yahoo' | 'emergency-call';
export type ReturnKeyTypeOptions = ReturnKeyType | ReturnKeyTypeAndroid | ReturnKeyTypeIOS;
```
storybook `options` takes array as an argument. So we have to convert it to:
```
const ReturnKeyType = ['done', 'go', 'next', 'search', 'send']
const ReturnKeyTypeAndroid = ['none', 'previous']
const ReturnKeyTypeIOS = ['default', 'google', 'join', 'route', 'yahoo', 'emergency-call']
export const ReturnKeyTypeOptions = [...ReturnKeyType, ...(isIOS ? ReturnKeyTypeIOS : ReturnKeyTypeAndroid)]
```

**Comment by dannyhw** [comment]
In the current web storybook solution there are some arg types inferred from type definitions however I'm not sure how intelligent it is or if it will work for RN, in theory some of it could work in the current beta but im not sure.
